backend/src/Edna/Upload/Service.hs

```diff
@@ -130,11 +133,23 @@ uploadFile' projSqlId@(SqlId proj) methodSqlId@(SqlId method)
       measurementsToSummary fileMeasurements
 
 insertTarget :: Text -> Edna (Text, TargetId)
-insertTarget targetName = (targetName,) . fromSqlSerial . tTargetId <$> LQ.insertTarget targetName
+insertTarget targetName = do
+  (targetName,) . fromSqlSerial . tTargetId <$> LQ.insertTarget targetName
+
+
+buildMdeLink :: Maybe MdeHost -> Text -> Maybe Text
+buildMdeLink Nothing _ = Nothing
```

Comment:
If below `host` is `Nothing`, you will get `Nothing` in result. Did you write this case as an optimization to avoid checking whether the string ends with a number in case `host` is `Nothing`? If yes, then it seems to make sense.

Also instead of pattern matching you can do `<- host` inside `Maybe` monad. Something like:
```

buildMdeLink mHost compoundName = do
  host <- mHost
  let (_, ns) = T.break isNumber compoundName
  guard (not $ T.null ns && T.all isNumber ns)
  Just . (<> "/" <> ns) . T.pack $ host
```

This way you will check whether `host` is `Just` only once.

Reply:
>  Did you write this case as an optimization to avoid checking whether the string ends with a number in case host is Nothing?

yes

> This way you will check whether host is Just only once.

you are right, I'll change it